bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -64,17 +62,19 @@ private CheckRun(CheckWorkItem workItem, PullRequest pr, PullRequestInstance prI
         this.labels = new HashSet<>(labels);
         this.newLabels = new HashSet<>(labels);
         this.censusInstance = censusInstance;
-        this.blockingLabels = blockingLabels;
-        this.issueProject = issueProject;
     }
 
     static void execute(CheckWorkItem workItem, PullRequest pr, PullRequestInstance prInstance, List<Comment> comments,
-                        List<Review> allReviews, List<Review> activeReviews, Set<String> labels, CensusInstance censusInstance, Map<String, String> blockingLabels,
-                        IssueProject issueProject) {
-        var run = new CheckRun(workItem, pr, prInstance, comments, allReviews, activeReviews, labels, censusInstance, blockingLabels, issueProject);
+                        List<Review> allReviews, List<Review> activeReviews, Set<String> labels, CensusInstance censusInstance) {
+        var run = new CheckRun(workItem, pr, prInstance, comments, allReviews, activeReviews, labels, censusInstance);
         run.checkStatus();
     }
 
+    private boolean checkTargetBranch() {
+        var matcher = workItem.bot.allowedTargetBranches().matcher(pr.targetRef());
```

Comment:
Should we use a pattern or should we simply just enumerate the protected branches? It seems potentially overkill to support a regex, it seems like an anti-pattern to have a repository with so many protected branches that they are cumbersome to enumerate, so you need a regex :smiley: